resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterManager.scala

```diff
@@ -33,7 +33,9 @@ private[spark] class KubernetesClusterManager extends ExternalClusterManager wit
   override def canCreate(masterURL: String): Boolean = masterURL.startsWith("k8s")
 
   override def createTaskScheduler(sc: SparkContext, masterURL: String): TaskScheduler = {
-    if (masterURL.startsWith("k8s") && sc.deployMode == "client") {
+    if (masterURL.startsWith("k8s") &&
+      sc.deployMode == "client" &&
+      !sc.conf.contains(KUBERNETES_EXECUTOR_POD_NAME_PREFIX)) {
```

Comment:
Is this sufficient to prevent use the client mode from end users? What about adding a special key when calling `spark-submit` in the driver and test that key here instead?

Reply:
I believe that logic might be beyond the scope of this PR. But I could add that if it seems appropriate.

Reply:
I think it's safer to have a new internal config key that is only used for this purpose. Checking the presence of `KUBERNETES_EXECUTOR_POD_NAME_PREFIX ` isn't sufficient as it is set by the submission client code that will be called in client mode also.